samples/msal-node-samples/auth-code-distributed-cache/src/AuthProvider.ts

```diff
@@ -0,0 +1,198 @@
+         * The state parameter can also be used to encode information of the app's state before redirect.
+         * You can pass the user's state in the app, such as the page or view they were on, as input to this parameter.
+         */
+        const state = this.cryptoProvider.base64Encode(
+            JSON.stringify({
+                csrfToken: this.cryptoProvider.createNewGuid(), // create a GUID for csrf
+                redirectTo: requestRoute || '/',
+            })
+        );
+
+        const msalInstance = this.getMsalInstance(sessionId);
+
+        try {
+            const authCodeUrl = await msalInstance.getAuthCodeUrl({
+                ...authCodeUrlParameters,
+                state,
+            });
+
+            return { authCodeUrl, state };
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async getTokenInteractive(tokenRequest: AuthorizationCodeRequest, authCodePayLoad: AuthorizationCodePayload, sessionId: string): Promise<AuthenticationResult | null> {
+        const msalInstance = this.getMsalInstance(sessionId);
+        let tokenResponse = null;
+
+        try {
+            performance.mark("acquireTokenByCode-start");
+            await msalInstance.getTokenCache().getAllAccounts(); // required for triggering beforeCacheACcess
+            tokenResponse = await msalInstance.acquireTokenByCode(tokenRequest, authCodePayLoad);
+            performance.mark("acquireTokenByCode-end");
+            performance.measure("acquireTokenByCode", "acquireTokenByCode-start", "acquireTokenByCode-end");
+        } catch (error) {
+            throw error;
+        }
+
+        return tokenResponse;
+    }
+
+    async getTokenSilent(silentRequest: SilentFlowRequest, sessionId: string): Promise<AuthenticationResult | null> {
+        const msalInstance = this.getMsalInstance(sessionId);
+        let tokenResponse = null;
+
+        try {
+            performance.mark("acquireTokenSilent-start");
+            await msalInstance.getTokenCache().getAllAccounts(); // required for triggering beforeCacheACcess
+            tokenResponse = await msalInstance.acquireTokenSilent(silentRequest);
+            performance.mark("acquireTokenSilent-end");
+            performance.measure("acquireTokenSilent", "acquireTokenSilent-start", "acquireTokenSilent-end");
+        } catch (error) {
+            throw error;
+        }
+
+        return tokenResponse;
+    }
+
+    private static async getMetadata(msalConfig: Configuration, cacheClient: RedisClientType): Promise<Configuration> {
+        const msalConfigWithMetadata = msalConfig;
+        const clientId = msalConfigWithMetadata.auth.clientId;
+        const tenantId = msalConfigWithMetadata.auth.authority!.split("/").pop()!;
+
+        try {
+            let [cloudDiscoveryMetadata, authorityMetadata] = await Promise.all([
+                cacheClient.get(`${clientId}.${tenantId}.discovery-metadata`),
```

Comment:
I'd say we should keep authority-metadata, since the config parameter for the CCA where this is used is `authorityMetadata` https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/configuration.md#auth-config-options